bat/ssh.go

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package bat
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+)
+
+//SSHClient contains details of the server to be connect
+type SSHClient struct {
+	ClientIP      string
+	ClientPort    string
+	ClientUser    string
+	ClientPass    string
+	ClientTimeout int64
+}
+
+func dialSSH(ctx context.Context, network, addr string, config *ssh.ClientConfig) (*ssh.Client, error) {
+	dialer := net.Dialer{
+		Timeout: config.Timeout,
+	}
+	conn, err := dialer.DialContext(ctx, network, addr)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to connect to ssh server: %v", err)
+	}
+	c, chs, r, err := ssh.NewClientConn(conn, addr, config)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create new SSH client connection : %v", err)
+	}
+	return ssh.NewClient(c, chs, r), nil
+}
+
+//createSSHClient creates a SSH session for the server specified
+//
+//Receive a struct with the details of the server that will be connected
+//create a tcp session that will be use to send and receive data
+//
+//session sould be created succesfully and wait for commands
+func createSSHClient(ctx context.Context, c SSHClient) (*ssh.Client, error) {
+	//If no port is defined, it will be default to 22
+	if c.ClientPort == "" {
+		c.ClientPort = "22"
+	}
+	if c.ClientTimeout == 0 {
+		c.ClientTimeout = 5
+	}
+	ClientFullAddress := fmt.Sprint(c.ClientIP + ":" + c.ClientPort)
```

Comment:
Best to use net.JoinHostPort 

https://golang.org/pkg/net/#JoinHostPort

Otherwise the URL will be invalid if ClientIP is IPv6